docs/onboarding.md

```diff
@@ -0,0 +1,158 @@
+  "public-name":"Alice Martin"
+}
+```
+
+If the user is logged in, display all instance information, except password
+#### Request
+```http
+GET https://alice.example.com/instance
+Cookie sessionid=xxxx
+```
+#### Response
+```json
+{
+  "_rev":"3-56521545485448482",
+  "domain": "alice.example.com",
+  "owner-email": "alice@example.com",
+  "public-locale":"fr",
+  "public-name":"Alice Martin"
+}
+```
+
+### POST /instance
+
+If the user is not logged in, allow to set the password. This action requires the `registerToken` defined on instance creation.*
+Once used, the `registerToken` is destroyed and cannot be used anymore.
+
+#### Request
+```http
+POST https://alice.example.com/instance
+Content-type: application/x-www-form-urlencoded
+registerToken=4256595659594842&password=my-secret
+```
+
+#### Response
+```
+301 Redirect
+Set-Cookie: sessionid=xxxx .example.com
+Location: onboarding.alice.example.com
+```
+
+If the user is logged in, allow to set instance fields
+
+#### Request
+```http
+POST https://alice.example.com/instance
+Content-type: application/json
+Cookie sessionid=xxxxx
+Authorization: base64(onboarding:onboardingapptoken)
+```
+```json
+{
+  "_rev":"3-56521545485448482",
+  "tz": "Europe/Berlin"
+}
+```
+
+#### Response
+```
+200 OK
+Content-type: application/json
+Cookie sessionid=xxxxx
+Authorization: base64(onboarding:onboardingapptoken)
+```
+```json
+{
+  "_rev":"4-15454854484828845221285",
+  "tz": "Europe/Berlin",
+  "domain": "alice.example.com",
+  "owner-email": "alice@example.com",
+  "public-locale":"fr",
+  "public-name":"Alice Martin"
+}
+```
+
+## Flow Example
+
+- The server administrator Bob creates an instance through the CLI. He knows the instance should be in french for an user named `alice`.
+```
+cozy-stack instance add alice.example.com --locale fr --public-name "Alice Martin" --tz Europe/Paris
+>> https://alice.cozycloud.cc?registerToken=42456565213125454842
+```
+The instance is created
+```json
+{
+  "domain": "alice.example.com",
+  "publicName": "Alice Martin",
+  "tz": "Europe/Paris",
+  "locale":"fr"
+}
+```
+- Eve knows Alice just had an instance created, she goes to `https://alice.cozycloud.cc`. There is no `registerToken`, so she only see a message (in french) along the lines of "This is the cozy for Alice Martin, this register link is incorrect, if you are Alice Martin please ask your sysadmin for a new link".
+- Alice navigates to `https://alice.cozycloud.cc?registerToken=42...42`
+- She is redirected to the `onboarding` application
+- The `onboarding` application receive the registerToken. It is the default onboarding application and therefore display the cozy cloud agreement and then ask for a Password.
+- The `onboarding` application use its `registerToken` to register the password. Registering the password automatically log Alice in and redirect her back to the `onboarding` app but with its normal `data-cozy-token`.
+- The `onboarding` application can then do whatever it needs to do :
```

Comment:
We should not forget to have CSRF protection from here. The `registerToken` should protect us on the creation of the instance with the password, but for the next steps we should have another anti-csrf token to prevent an attacker from registering unwanted informations into the account (even if it does not known the password, it could forge requests to put its own email for instance).

Event though, this may be part of another "global action" against CSRF more than this specific document, it can be specified here..